repository/src/test/java/org/alfresco/repo/search/impl/querymodel/impl/db/DBQueryEngineSelectNodesWithPermissionTest.java

```diff
@@ -0,0 +1,196 @@
+    
+    @Autowired
+    private NodeService nodeService;
+    
+    @Autowired
+    private ServiceRegistry serviceRegistry;
+    
+    @Autowired
+    private PermissionService permissionService;
+    
+    @Autowired
+    private AuthenticationComponent authenticationComponent;
+    
+    private NodeRef rootNodeRef;
+    private List<NodeRef> testData = new ArrayList<>();
+    private String prefix;
+    private DBQueryEngine queryEngine;
+    
+    @Before
+    public void setup()
+    {
+//        queryEngine =(DBQueryEngine) this.applicationContext.getBean("dbQueryEngineImpl");
+//        permissionService = Mockito.spy(permissionService);
+//        queryEngine.setPermissionService(permissionService);
+        
+//        AclDAO aclDAO = (AclDAO) applicationContext.getBean("aclDAO");
+//        
+//        ((PermissionServiceImpl)permissionService).setAclDAO(aclDAO);
+        
+        prefix = "doc_"+System.currentTimeMillis() + "_";
+        authenticationComponent.setSystemUserAsCurrentUser();
+        
+//        StoreRef storeRef = nodeService.createStore(StoreRef.PROTOCOL_WORKSPACE, "Test_" + System.currentTimeMillis());
+//        St
+//        rootNodeRef = nodeService.getRootNode(storeRef);
+       
+        AuthenticationUtil.runAsSystem(new RunAsWork<Integer>()
+        {
+            public Integer doWork() throws Exception {
+                createTestData(TEST_DATA_AMOUNT);
+                return 0;
+            }
+        });
+    }
+    
+    @After
+    public void teardown()
+    {
+        clearTestData();
+        AuthenticationUtil.clearCurrentSecurityContext();
+    }
+    
+    private void clearTestData()
+    {
+        for (NodeRef data: this.testData)
+        {
+            retryingTransactionHelper.doInTransaction(() -> 
+            {
+                nodeService.deleteNode(data);
+                return null;
+            });
+        }
+        
+        testData.clear();
+    }
+
+    private void createTestData(int amount)
+    {
+        for(int i = 0; i < amount; i++)
+        {
+            testData.add(createNode(prefix+i));
+        }
+
+    }
+
+    private NodeRef createNode(String title)
+    {
+        PropertyMap propertyMap = new PropertyMap();
+        propertyMap.put(ContentModel.PROP_NAME, title);
+        
+        NodeRef nodeRef =  retryingTransactionHelper.doInTransaction(() -> nodeService.createNode(
+                rootNodeRef,
+                ContentModel.ASSOC_CHILDREN,
+                QName.createQName(TEST_NAMESPACE, GUID.generate()),
+                ContentModel.TYPE_CONTENT,
+                propertyMap).getChildRef());
+
+        return nodeRef;
+        
+    }
+    
+    @Test
+    public void shouldSelectNodesWhenExecutingQueryAsSystemUser() 
+    {
+        String query = "select * from cmis:document as d where d.cmis:name like '%"+prefix+"%'";
+//        when(permissionService.getReaders(anyLong())).thenReturn(Collections.emptySet());
+        
```

Comment:
Please remove unused code

Reply:
My bad this class was not supposed to be committed.